bats/tests/k8s/up-downgrade-k8s.bats

```diff
@@ -0,0 +1,140 @@
+# Test case 8 & 13
+
+# shellcheck disable=SC2030,SC2031
+# SC2030 (info): Modification of output is local (to subshell caused by @bats test).
+# SC2031 (info): output was modified in a subshell. That change might be lost.
+
+setup() {
+    load '../helpers/load'
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start rancher desktop' {
+    start_kubernetes
+    wait_for_apiserver
+    # the docker context "rancher-desktop" may not have been written
+    # even though the apiserver is already running
+    wait_for_container_engine
+}
+
+@test 'deploy nginx - always restart' {
+    ctrctl pull nginx
+    run ctrctl run -d -p 8585:80 --restart=always --name nginx-restart nginx
+    assert_success
+}
+
+@test 'deploy nginx - no restart' {
+    run ctrctl run -d -p 8686:80 --restart=no --name nginx-no-restart nginx
+    assert_success
+}
+
+@test 'deploy busybox' {
+    run kubectl_exe create deploy busybox --image=busybox --replicas=2 -- /bin/sh -c "sleep inf"
+    assert_success
+}
+
+verify_nginx() {
+    local nginx_ports=(8585 8686)
+    for port in "${nginx_ports[@]}"; do
+        run curl http://localhost:"$port"
+        assert_success
+        assert_output --partial "Welcome to nginx!"
+    done
+}
+
+@test 'verify nginx before upgrade' {
+    try verify_nginx
+}
+
+verify_busybox() {
+    run kubectl get pods --selector="app=busybox" -o jsonpath='{.items[*].status.phase}'
+    assert_output --partial "Running Running"
+}
+
+@test 'verify busybox before upgrade' {
+    try verify_busybox
+}
+
+verify_images() {
+    if using_docker; then
+        run ctrctl images
+        assert_output --partial "nginx" "busybox"
+    else
+        run ctrctl images --format json
+        assert_output --partial "\"Repository\":\"nginx"
```

Comment:
Since there is no variable interpolation, maybe use single quotes, so you don't need to escape all the internal double quotes?

```suggestion
        assert_output --partial '"Repository":"nginx'
```